schoolmatch.py

```diff
@@ -45,146 +35,100 @@ def string_truncate(strings_to_truncate, words_to_remove, truncated_list_name):
             modifying_string = modifying_string.strip()
         truncated_list_name.append(modifying_string)  
     return truncated_list_name
-#Will take a look at stop words next
+
+
+def school_match(school_type, output_type, rs_df, ospi_df):
+    '''
+    Takes the type of school (public or independent) as a string, a string 
+    "matched" or "unmatched" that determines whether the output is the schools 
+    that matched or those that didn't, the processed dataframe from 
+    Rainier Scholars, and the processed dataframe from OPSI using only the 
+    code, name, and city columns in that order and looks for matches by 
+    comparing the truncated names.
+    '''
+    ospi_code = []

```

Comment:
what you are doing in this function is filtering down to subsets of both data frames that have the same city, and then matching on TruncName.

If you flip this order of operations around, you can match on TruncName (using merge) and then filter for rows where City == OSPI City.  

all_pub_df = pd.merge(rs_df, ospi_pub_df, 
                        how='left', left_on='TruncName', right_on='TruncName', 
                        suffixes=['', '_OSPI'])
all_matched_df = all_pub_df[all_pub_df["City"] == all_pub_df["City_OSPI"]]
You will need some extra code here to fix up the column names.  

Then, to create the unmatched df, you find everything in rs_df that isn't in all_matched_df
all_unmatched_df = rs_pub_df[~rs_pub_df["Name"].isin(all_matched_df["Name"])]